cpp/src/arrow/util/io_util.cc

```diff
@@ -97,8 +97,16 @@
 #include "arrow/util/logging.h"
 
 // For filename conversion
-#if defined(_WIN32)
+#ifdef _WIN32
```

Comment:
This does not really fall into the "filename conversion" category, can you move these additions up?